### PR TITLE
docs - add GUC gp_resource_group_bypass

### PR DESCRIPTION
**Pull request by mkiyama**
Will be ported to 5X_STABLE

Link to HTML on GPDB review doc site.
http://docs-gpdb-review-staging.cfapps.io/review/ref_guide/config_params/guc-list.html#gp_resource_group_bypass

GUC list by category
http://docs-gpdb-review-staging.cfapps.io/review/ref_guide/config_params/guc_category-list.html#topic444

Dev PR https://github.com/greenplum-db/gpdb/pull/5205

### CODE REVIEW

**Comment by mkiyama** [comment]
@gaos1 @dujialun @pivotal-ning-yu 
Could somebody please review.

Thanks for your help.

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 1]
```diff
@@ -4854,6 +4855,44 @@
       </table>
     </body>
   </topic>
+  <topic id="gp_resource_group_bypass">
+    <title>gp_resource_group_bypass</title>
+    <body>
+      <note>The <codeph>gp_resource_group_bypass</codeph> server configuration parameter is enforced
+        only when resource group-based resource management is active.</note>
+      <p>Enables or disables enforcement of resource group limits on Greenplum Database resources.
```
Comment (gaos1) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: better to emphasis that this is only bypass concurrency limit. e.g. change "Enables or disables enforcement of resource group limits on Greenplum Database resources." to "Enables or disables enforcement of resource group concurrency limits on Greenplum Database resources."

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 2]
```diff
@@ -4854,6 +4855,44 @@
...
+        Resource groups manage resources such as CPU, memory, and the number of concurrent
+        transactions that are used by queries and external components such as PL/Container. Default
+        value is <codeph>true</codeph>, enable enforcement of resource group limits. </p>
```
Comment (gaos1) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: default value is false

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 3]
```diff
@@ -4854,6 +4855,44 @@
...
+      <p>You can set this parameter to <codeph>false</codeph> to bypass resource group resource
+        limitations so that a query can run immediately. For example, you can set the parameter to
+          <codeph>false</codeph> for a session to run a query against the system catalog or a
```
Comment (gaos1) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: you can set the parameter to true .....

Comment (gaos1) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: system catalog --> system catalog query?

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 4]
```diff
@@ -4854,6 +4855,44 @@
...
+        similar query that requires a minimal amount of resources.</p>
+      <p>This parameter can be set for a database system, an individual database, or a session or
```
Comment (gaos1) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: the parameter can only be set on session.

**Comment by gaos1** [comment]
We might also need to highlight the following points:
1) bypassed queries will still run inside a resource group. The resource group assignment is the same as normal queries.
2) bypassed queries will be limited to have a fixed 1M memory quota. The actual used memory will be allocated from resource group shared memory or global shared memory.  If there is no available shared memory to meet the memory allocation request from bypassed queries, these bypassed queries will fail. 
3) The GUC cannot be set inside a transaction or a function.

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 5]
```diff
@@ -4861,10 +4861,14 @@
       <note>The <codeph>gp_resource_group_bypass</codeph> server configuration parameter is enforced
         only when resource group-based resource management is active.</note>
       <p>Enables or disables the enforcement of resource group concurrent transaction limits on
-        Greenplum Database resources. The default value is <codeph>false</codeph>, enable the
-        enforcement of resource group limits. Resource groups manage resources such as CPU, memory,
-        and the number of concurrent transactions that are used by queries and external components
-        such as PL/Container. </p>
+        Greenplum Database resources. The default value is <codeph>false</codeph>, enforce of
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: enforce of -> enforce

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 6]
```diff
@@ -4854,6 +4855,53 @@
       </table>
     </body>
   </topic>
+  <topic id="gp_resource_group_bypass">
+    <title>gp_resource_group_bypass</title>
+    <body>
+      <note>The <codeph>gp_resource_group_bypass</codeph> server configuration parameter is enforced
+        only when resource group-based resource management is active.</note>
+      <p>Enables or disables the enforcement of resource group concurrent transaction limits on
+        Greenplum Database resources. The default value is <codeph>false</codeph>, enforce of
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: enforce of -> which enforces

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 7]
```diff
@@ -4854,6 +4855,53 @@
...
+        resource group transaction limits. Resource groups manage resources such as CPU, memory, and
+        the number of concurrent transactions that are used by queries and external components such
+        as PL/Container. </p>
+      <p>You can set this parameter to <codeph>true</codeph> to bypass resource group resource
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: resource group resource -> resource group

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 8]
```diff
@@ -4854,6 +4855,53 @@
...
+        set the parameter to <codeph>true</codeph> for a session to run a system catalog query or a
+        similar query that requires a minimal amount of resources. </p>
+      <p>When you set this parameter to <codeph>true</codeph> and a run a query, the query runs in
+        this environment:<ul id="ul_vgc_rw4_j2b">
+          <li>The query runs inside a resource group. The resource group assignment for the query
+            does not change.</li>
+          <li>The query memory quota is 1MB. The memory is allocated from resource group shared
+            memory or global shared memory. The query fails if there is not enough shared memory
+            available to fulfill the memory allocation request.</li>
+        </ul></p>
```
Comment (pivotal-ning-yu) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: This description might be different with current implementation. A bypassed query runs inside a resgroup w/o resource limitation, there is not a 1MB limitation yet. @gaos1 could I double check with you that we want to put a hard limitation on bypassed queries? If yes then we need some code changes.

Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: @gaos1 @pivotal-ning-yu 
Will the 1 MB limitation be in this release?

thanks

Comment (gaos1) [reply gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: @mkiyama we will introduce a memory limitation for bypassed queries. the exact amount is under testing and we will update you soon.  You can hold this doc PR a while. Thanks.

**Comment by mkiyama** [comment]
@gaos1 @pivotal-ning-yu @dujialun 
Updated GUC information based on email w/dev updates

**Pull request merged**